service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/StructuredObservationValueMapper.java

```diff
@@ -12,30 +13,24 @@
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Component;
 
-import com.google.common.collect.ImmutableMap;
-
 import lombok.RequiredArgsConstructor;
+import uk.nhs.adaptors.gp2gp.ehr.mapper.parameters.InterpretationCodeTemplateParameters;
+import uk.nhs.adaptors.gp2gp.ehr.mapper.parameters.ReferenceRangeTemplateParameters;
+import uk.nhs.adaptors.gp2gp.ehr.utils.TemplateUtils;
 
 @RequiredArgsConstructor(onConstructor = @__(@Autowired))
 @Component
 public class StructuredObservationValueMapper {
-    private static final Map<Class<? extends IBaseElement>, Function<IBaseElement, String>> VALUE_MAPPING_FUNCTIONS =
-        ImmutableMap.of(Quantity.class, value -> ObservationValueQuantityMapper.processQuantity((Quantity) value),
-            StringType.class, value -> processStringType((StringType) value));
-    private static final String STRING_VALUE_TEMPLATE = "<value xsi:type=\"ST\">%s</value>";
-    private static final String REFERENCE_RANGE_TEMPLATE = "<referenceRange typeCode=\"REFV\">"
-        + "<referenceInterpretationRange classCode=\"OBS\" moodCode=\"EVN.CRT\">"
-        + "<text>%s</text>"
-        + "<value>%s</value>"
-        + "</referenceInterpretationRange>"
-        + "</referenceRange>";
-    private static final String LOW_RANGE_TEMPLATE = "<low value=\"%s\"/>";
-    private static final String HIGH_RANGE_TEMPLATE = "<high value=\"%s\"/>";
-    private static final String INTERPRETATION_CODE_TEMPLATE = "<interpretationCode code=\"%s\" "
-        + "codeSystem=\"2.16.840.1.113883.2.1.6.5\" "
-        + "displayName=\"%s\">"
-        + "<originalText>%s</originalText>"
-        + "</interpretationCode>";
+    private static final Map<Class<? extends IBaseElement>, Function<IBaseElement, String>> VALUE_MAPPING_FUNCTIONS = Map.of(
+        Quantity.class, value -> ObservationValueQuantityMapper.processQuantity((Quantity) value),
+        StringType.class, value -> processStringType((StringType) value)
+    );
+    private static final Mustache STRING_VALUE_TEMPLATE = TemplateUtils.compileTemplate("<value xsi:type=\"ST\">{{value}}</value>");
+    private static final Mustache REF_RANGE_TEMPLATE = TemplateUtils.loadTemplate("ehr_reference_range_template.mustache");
+
+    private static final Mustache LOW_RANGE_TEMPLATE = TemplateUtils.compileTemplate("<low value=\"{{{low}}}\"/>");
```

Comment:
Just a thought, we could just use a `.` here, then we don't have to create a Map to populate the value
```suggestion
    private static final Mustache LOW_RANGE_TEMPLATE = TemplateUtils.compileTemplate("<low value=\"{{{.}}}\"/>");
```

Reply:
Nice!